db/database.go

```diff
@@ -837,66 +862,66 @@ func (db *DatabaseContext) DeleteUserSessions(userName string) error {
 // When compact is run, Sync Gateway initiates a normal delete operation for the document and xattr (a Sync Gateway purge).  This triggers
 // removal of the document from the index.  In the event that the document has already been purged by server, we need to recreate and delete
 // the document to accomplish the same result.
-func (db *Database) Compact() (int, error) {
+func (context *DatabaseContext) Compact() (int, error) {
 
-	if !atomic.CompareAndSwapUint32(&db.CompactState, DBCompactNotRunning, DBCompactRunning) {
+	if !atomic.CompareAndSwapUint32(&context.CompactState, DBCompactNotRunning, DBCompactRunning) {
 		return 0, base.HTTPErrorf(http.StatusServiceUnavailable, "Compaction already running")
 	}
 
-	defer atomic.CompareAndSwapUint32(&db.CompactState, DBCompactRunning, DBCompactNotRunning)
+	defer atomic.CompareAndSwapUint32(&context.CompactState, DBCompactRunning, DBCompactNotRunning)
 
 	// Compact should be a no-op if not running w/ xattrs
-	if !db.UseXattrs() {
+	if !context.UseXattrs() {
 		return 0, nil
 	}
 
 	// Trigger view compaction for all tombstoned documents older than the purge interval
-	purgeIntervalDuration := time.Duration(-db.PurgeInterval) * time.Hour
+	purgeIntervalDuration := time.Duration(-context.PurgeInterval) * time.Hour
 	startTime := time.Now()
 	purgeOlderThan := startTime.Add(purgeIntervalDuration)
-	results, err := db.QueryTombstones(purgeOlderThan)
+	results, err := context.QueryTombstones(purgeOlderThan)
 	if err != nil {
 		return 0, err
 	}
 
-	ctx := db.Ctx
-
-	base.InfofCtx(ctx, base.KeyAll, "Compacting purged tombstones for %s ...", base.UD(db.Name))
+	base.Infof(base.KeyAll, "Database %s: Compacting purged tombstones...", base.MD(context.Name))
 	purgeBody := Body{"_purged": true}
 	purgedDocs := make([]string, 0)
 
 	var tombstonesRow QueryIdRow
 	for results.Next(&tombstonesRow) {
-		base.InfofCtx(ctx, base.KeyCRUD, "\tDeleting %q", tombstonesRow.Id)
+		base.Infof(base.KeyCRUD, "Database %s:\tDeleting %q", base.MD(context.Name), tombstonesRow.Id)
```

Comment:
We're losing context logging here now we've moved to `DatabaseContext`. It feels like a step backwards.

As suggested on the call, if we have the scheduler create a `Database`, we could keep context here and in Purge.